### PR TITLE
KFSPTS-8016 fix code that updates report status in Concur and add mor…

### PR DESCRIPTION
**Pull request by db483**
…e logging info

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/concur/rest/xmlObjects/ExpenseReportDetailsDTO.java [hunk 1]
```diff
@@ -12,6 +12,9 @@
 @XmlAccessorType(XmlAccessType.NONE)
 public class ExpenseReportDetailsDTO {
     
+    @XmlElement(name = "ReportId")
+    private String reportId;
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/concur/rest/xmlObjects/ExpenseReportDetailsDTO.java]: This area uses "ReportId" and "reportId", but the ConcurReport class uses "reportID" instead. (The other XML DTO is also using "RequestID" and "requestID".) Does this DTO need to capitalize the "D" as well, or does Concur or KFS expect it to be just "Id" and not "ID"?

File: src/main/java/edu/cornell/kfs/concur/service/impl/ConcurReportsServiceImpl.java [hunk 2]
```diff
@@ -197,9 +204,32 @@ private String buildWorkflowUpdateXML(String workflowURI, String action, String
...
+    private String addMessageHeader(String message) {
+        if (message.length() > 0) {
+            message = ConcurConstants.ERROR_MESSAGE_HEADER + message;
+        }
+        return message;
+    }
+
+    private String truncateMessageLength(String message, int maxLength) {
+        if (message.length() > maxLength) {
+            message = message.substring(0, maxLength + 1);
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/concur/service/impl/ConcurReportsServiceImpl.java]: This line will actually return a substring of length maxLength + 1. If the substring indeed needs to be no longer than maxLength, then replace "maxLength + 1" with just "maxLength".

File: src/main/java/edu/cornell/kfs/concur/rest/xmlObjects/ExpenseReportDetailsDTO.java (hunk 1 above)
Comment (db483) [inline src/main/java/edu/cornell/kfs/concur/rest/xmlObjects/ExpenseReportDetailsDTO.java]: good point. Concur expects ReportId in this case.

File: src/main/java/edu/cornell/kfs/concur/service/impl/ConcurReportsServiceImpl.java (hunk 2 above)
Comment (db483) [inline src/main/java/edu/cornell/kfs/concur/service/impl/ConcurReportsServiceImpl.java]: good catch, the +1 needs to be removed.

**Comment by db483** [comment]
I fixed the truncate method. I would like to leave the reportId property as is although there is some inconsistency. Concur uses different naming for different reports.

**Comment by db483** [comment]
This is ready for review again.

**Pull request merged**